packages/teleport/src/Discover/Shared/JoinTokenContext.tsx

```diff
@@ -60,15 +64,22 @@ interface PromiseResult {
 }
 
 let abortController: AbortController;
-let result: PromiseResult;
+let cachedJoinTokenResult: PromiseResult;
+
+export function clearCachedJoinTokenResult() {
```

Comment:
```suggestion
function clearCachedJoinTokenResult() {
```

I don't think we need to export this

Reply:
i require it in several places, in storybook with msw (results get cached so i need to clear it between different story states that makes http requests), and when i provide a CatchError, `onRetry` i also need to clear the cache, it looks something like this (in my next PR). 

```ts
    <CatchError
      onRetry={clearCachedJoinTokenResult}
      fallbackFn={props => (
        <Box>
          <Heading />
          <ScriptBox pollState="error">
            <TextIcon mt={2} mb={3}>
              <Icons.Warning ml={1} color="danger" />
              Encountered Error: {props.error.message}
            </TextIcon>
            <ButtonBlueText ml={2} onClick={props.retry}>
              Refetch a command
            </ButtonBlueText>
          </ScriptBox>
        </Box>
      )}
    >
      <Suspense
        fallback={
          <Box>
            <Heading />
            <ScriptBox height="144px">
              <Box textAlign="center" height="108px">
                <Indicator />
              </Box>
            </ScriptBox>
          </Box>
        }
      >
        <DownloadScript {...props} />
      </Suspense>
    </CatchError>
```

Reply:
Ah sorry I missed those usages - I must not have loaded all of the diff

Reply:
no worries! it's because these usages are going to be on my next PR (it's not in this PR) i wanted to make my PR small